src/Oauth1.php

```diff
@@ -239,8 +239,12 @@ private function prepareParameters($data)
      */
     private function signUsingHmacSha1($baseString)
     {
-        $key = rawurlencode($this->config['consumer_secret'])
-            . '&' . rawurlencode($this->config['token_secret']);
+        $key = rawurlencode($this->config['consumer_secret']);
+
+        // Add token only if present to avoid wrong encoding due to the superflous ampersand (&)
+        if (isset($this->config['token_secret']) && !empty($this->config['token_secret'])) {
```

Comment:
I've closed this "dirty" pull request and submitted a new one, using only `empty` and without the "exceding" commits. Hope this is good...